### PR TITLE
[APM-CI][Integration test][Ruby] Support JRuby with different JDK versions

### PR DESCRIPTION
**Pull request by v1v**
Caused by https://github.com/elastic/apm-agent-ruby/issues/287

## Highlights
- Support jruby with different jdk versions
- Enable docker login with a retry + sleep in case any infra issues.
- Ruby versions moved from `-` split to `:` as it is the way we can support our internal docker registry.
- Fix pipeline to bypass a weird script approval misleading message.

## Tasks
- [x] Test locally for the current list of versions in `spec/.jenkins_ruby.yml`
- [x] Validate whether the Jenkinsfile works as expected. [here](https://apm-ci.elastic.co/job/apm-agent-ruby/job/apm-agent-ruby-mbp/job/PR-403/14/)
- [x] --Script approval is required-- False! It's solved with https://github.com/elastic/apm-agent-ruby/pull/403/commits/db5ab60768dd3cf15f2e3321fd280af7bd6dedfe
- [x] Benchmark is also required. Unfortunately, the PRs are not validating it and I've just found it

## Todo
- I'd rather prefer to handle the stage naming in the UI Pipeline view in another PR as it does require to change the shared library too. What do I mean? See the below screenshot:
 
image

The name of the stage is based on the name of the docker images plus the framework, as the name of the new docker images are hosted in our internal docker registry, their names are quite long. 

What do you think?

### CODE REVIEW

**Comment by v1v** [comment]
jenkins run the tests please

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ Gemfile.lock
 spec/elastic_apm.log
 spec/ruby-agent-junit.xml
 .gem
+*.bulk
```
Comment (mikker) [inline .gitignore]: What's this?

Comment (v1v) [reply .gitignore]: That's a file auto-generated when running the benchmarking locally:
- https://github.com/elastic/apm-agent-ruby/pull/403/files#diff-964990e4e95f38885151096be29638e5R39

**Comment by v1v** [comment]
For the record, we will work on providing a nicer UI in the Jenkins pipeline to simplify the name of the stages, but that's a follow-up.

**Comment by mdelapenya** [comment]
Shaun approves 😎 

image

**Pull request merged**